common/utils/src/main/java/org/graalvm/buildtools/utils/NativeImageUtils.java

```diff
@@ -122,7 +122,8 @@ public static void checkVersion(String requiredVersion, String versionToCheck) {
         if (versionToCheck.contains("GraalVM Runtime Environment")) {
             return; // later than 22.3.1 (e.g., GraalVM for JDK 17 / GraalVM for JDK 20)
         }
-        if (versionToCheck.startsWith("GraalVM dev") || versionToCheck.startsWith("native-image dev")) {
+        if ((versionToCheck.startsWith("native-image") && versionToCheck.contains("-dev+")) ||
```

Comment:
I would change `-dev+` to just `dev`. We should try to be future-proof, i.e., we don't want to adapt this change any time the version string of GraalVM changes.

Maybe even reduce the whole check to just `if (versionToCheck.contains("dev") {`.